fenix/app/src/main/java/org/mozilla/fenix/translations/LoadingIndicator.kt

```diff
@@ -0,0 +1,153 @@
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.DEFAULT_MAX_LINES
+import org.mozilla.fenix.theme.FirefoxTheme
+
+@Composable
+@LightDarkPreview
+private fun DownloadIndicatorPreview() {
+    FirefoxTheme {
+        Column(
+            modifier = Modifier
+                .background(FirefoxTheme.colors.layer1)
+                .padding(16.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+        ) {
+            DownloadIndicator(
+                text = stringResource(id = R.string.translations_bottom_sheet_translating_in_progress),
+                icon = painterResource(id = R.drawable.ic_download_circle_spinner),
+            )
+        }
+    }
+}
+
+/**
+ * Download indicator.
+ *
+ * @param text The download indicator text to be displayed
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param backgroundColor The background [Color] of the download indicator.
+ * @param textColor [Color] to apply to the download indicator text.
+ * @param icon Optional [Painter] used to display an [Icon] before the download indicator text.
+ */
+@Composable
+fun DownloadIndicator(
+    text: String,
+    modifier: Modifier = Modifier,
+    backgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+    textColor: Color = FirefoxTheme.colors.textActionPrimary,
+    icon: Painter? = null,
+) {
+    BaseDownloadIndicator(
+        text = text,
+        textColor = textColor,
+        modifier = modifier,
+        icon = icon,
+        backgroundColor = backgroundColor,
+        iconModifier = Modifier
+            .rotate(rotationAnimation())
+            .wrapContentSize(),
+        tint = FirefoxTheme.colors.iconActionPrimary,
+    )
+}
+
+@Composable
+private fun BaseDownloadIndicator(
+    text: String,
+    textColor: Color,
+    modifier: Modifier = Modifier,
+    icon: Painter? = null,
+    backgroundColor: Color,
+    iconModifier: Modifier = Modifier,
+    tint: Color,
+) {
+    // Required to detect if font increased due to accessibility.
+    val fontScale: Float = LocalConfiguration.current.fontScale
+
+    Row(
```

Comment:
Since our use case for the download indicator is inside of a `PrimaryButton`, I think we should be looking to use a `Button` here instead of a `Row`. Would it make sense to replace this `Row` with the `Button` definition in https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt#54-61 for the time being?

Reply:
Actually, I am not fully correct here, we also have an use case with the loading indicator as a standalone list item. See https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7341%3A209640&mode=dev 

<img width="193" alt="Screen Shot 2023-10-06 at 11 53 09 AM" src="https://github.com/mozilla-mobile/firefox-android/assets/1190888/89920fe7-25a6-4843-ab5f-a6986e2ea78a">

Reply:
Here's some changes I am looking for:
- We should have a `LoadingIndicator` that displays the icon only (without the text) and handles the rotation animation as its own composable. This will be used in a `ListItem` as part of the `afterListAction` as seen in https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt#223. This will handle the use case in "Download languages" screen above and can also be reused in the Button use case.
- We should change your existing composable with the icon and text to be the Button use case. This means changing the `Row` to a `Button` as described earlier. I am specifically looking for you to split up the indicator to its own Composable that could be reuse for both use cases. I think we should make the button standalone for now instead of changing our existing [Button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt) to accept other composable `content` for the time being and think about it as a follow up after we land our first iteration of this button.

Reply:
Hi, I don't think it should be a button because it is not clickable and at accessibility it shouldn't say "button" . I will modify all to support the use case for download language screen .

Reply:
From figma it should be an indicator not a button

Reply:
Wouldn't the use case inside of the bottom sheet be a button? https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7202%3A99073&mode=dev

<img width="456" alt="Screen Shot 2023-10-06 at 12 20 12 PM" src="https://github.com/mozilla-mobile/firefox-android/assets/1190888/690585dc-d7e3-47eb-8fa3-7963c0cc26fa">

Reply:
We're also expecting this to be a button in the bottom as described in the accessibility notes in https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7224%3A198420&mode=dev

<img width="391" alt="Screen Shot 2023-10-06 at 12 22 11 PM" src="https://github.com/mozilla-mobile/firefox-android/assets/1190888/d3339bfd-6a90-45b7-8bfc-1c2e56b64536">

Reply:
To be clear, I am asking for 2 Composables:
- Button
- Indicator

Reply:
https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD?node-id=7221:186934&mode=design#571221606

Reply:
I understand two elements , but I'm not sure it should be a button .

Reply:
Done

Reply:
Thanks for the patch! I think the `Indicator` should have the role `ProgressBarRangeInfo`. I left some more info below.